File: spring-data-requery-test-java/src/test/java/org/springframework/data/requery/domain/time/ElementCollectionTest.java
package org.springframework.data.requery.domain.time;

import lombok.extern.slf4j.Slf4j;
import org.junit.Test;
import org.springframework.data.requery.domain.AbstractDomainTest;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

/**
 * @author Diego on 2018. 6. 12..
 */
@Slf4j
public class ElementCollectionTest extends AbstractDomainTest {

    @Test
    public void insert_element_collection() {

        assertThatThrownBy(() -> {
            ElementCollectionUser user = new ElementCollectionUser();
            user.setId(UUID.randomUUID());
            user.setName("user");
            Set<String> phoneNumbers = new HashSet<>();
            phoneNumbers.add("555-5555");
            phoneNumbers.add("666-6666");
            user.setPhoneNumbers(phoneNumbers);

            requeryTemplate.insert(user);

            ElementCollectionUser saved = requeryTemplate.select(ElementCollectionUser.class)
                .where(ElementCollectionUser.ID.eq(user.getId()).and(ElementCollectionUser.NAME.eq(user.getName())))
                .get()
                .first();

            assertThat(saved).isNotNull();
            assertThat(saved.getPhoneNumbers()).hasSize(2).containsOnly("555-5555","666-6666");
        }).isInstanceOf(ClassCastException.class);
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/time/ElementCollectionTest.kt
package com.coupang.springframework.data.requery.domain.time

import com.coupang.kotlinx.logging.KLogging
import com.coupang.springframework.data.requery.domain.AbstractDomainTest
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.Test
import java.util.*

/**
 * ElementCollectionTest
 *
 * @author debop@coupang.com
 * @since 18. 5. 31
 */
class ElementCollectionTest: AbstractDomainTest() {

    companion object: KLogging()

    @Test
    fun `insert element collection`() {

        assertThatThrownBy {
            val user = ElementCollectionUser().apply {
                id = UUID.randomUUID()
                name = "user"

                phoneNumbers = mutableSetOf("555-5555", "666-6666")
            }

            requeryTmpl.insert(user)


            val saved = dataStore
                .select(ElementCollectionUser::class.java)
                .where(ElementCollectionUser.ID.eq(user.id).and(ElementCollectionUser.NAME.eq(user.name)))
                .get()
                .first()

            assertThat(saved).isNotNull

            assertThat(saved.phoneNumbers).hasSize(2).containsOnly("555-5555", "666-6666")

        }.isInstanceOf(ClassCastException::class.java)
    }
}